engine/access/rpc/backend/backend_transactions_test.go

```diff
@@ -0,0 +1,333 @@
+package backend
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/dgraph-io/badger/v2"
+	"github.com/onflow/flow/protobuf/go/flow/access"
+	"github.com/onflow/flow/protobuf/go/flow/entities"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	acc "github.com/onflow/flow-go/access"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/state/protocol"
+	bprotocol "github.com/onflow/flow-go/state/protocol/badger"
+	"github.com/onflow/flow-go/state/protocol/util"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+func (suite *Suite) withPreConfiguredState(f func(snap protocol.Snapshot)) {
+	identities := unittest.CompleteIdentitySet()
+	rootSnapshot := unittest.RootSnapshotFixture(identities)
+	util.RunWithFullProtocolState(suite.T(), rootSnapshot, func(db *badger.DB, state *bprotocol.ParticipantState) {
+		epochBuilder := unittest.NewEpochBuilder(suite.T(), state)
+
+		epochBuilder.
+			BuildEpoch().
+			CompleteEpoch()
+
+		// get heights of each phase in built epochs
+		epoch1, ok := epochBuilder.EpochHeights(1)
+		require.True(suite.T(), ok)
+
+		// setup AtHeight mock returns for State
+		for _, height := range epoch1.Range() {
+			suite.state.On("AtHeight", epoch1.Range()).Return(state.AtHeight(height))
+		}
+
+		snap := state.AtHeight(epoch1.Range()[0])
+		suite.state.On("Final").Return(snap).Once()
+		suite.communicator.On("CallAvailableNode",
+			mock.Anything,
+			mock.Anything,
+			mock.Anything).
+			Return(nil).Once()
+
+		f(snap)
+	})
+
+}
+
+// TestGetTransactionResultReturnsUnknown returns unknown result when tx not found
+func (suite *Suite) TestGetTransactionResultReturnsUnknown() {
+	suite.withPreConfiguredState(func(snap protocol.Snapshot) {
+		block := unittest.BlockFixture()
+		tbody := unittest.TransactionBodyFixture()
+		tx := unittest.TransactionFixture()
+		tx.TransactionBody = tbody
+
+		coll := flow.CollectionFromTransactions([]*flow.Transaction{&tx})
+		suite.state.On("AtBlockID", block.ID()).Return(snap, nil).Once()
+
+		suite.transactions.
+			On("ByID", tx.ID()).
+			Return(nil, storage.ErrNotFound)
+
+		backend := New(
+			Params{
+				State:                suite.state,
+				CollectionRPC:        suite.colClient,
+				Blocks:               suite.blocks,
+				Transactions:         suite.transactions,
+				ExecutionReceipts:    suite.receipts,
+				ChainID:              suite.chainID,
+				AccessMetrics:        metrics.NewNoopCollector(),
+				MaxHeightRange:       DefaultMaxHeightRange,
+				Log:                  suite.log,
+				SnapshotHistoryLimit: DefaultSnapshotHistoryLimit,
+				Communicator:         suite.communicator,
+			})
```

Comment:
nit: you could add a `defaultParams()` method which returns this default set. That way you don't need to repeat this block of code in each test. If you need to use a specific value for some fields, you can just update the param set before creating the backend